docs/polygon_draw.js

```diff
@@ -366,22 +387,79 @@ function setUpPolygonDrawing(map, firebasePromise) {
  * @param {google.maps.Map} map Map to display regions on
  * @param {Promise<any>} firebasePromise Promise with Firebase damage data (also
  *     implies that authentication is complete)
- * @return {Promise} promise that is resolved when all initialization is done
- *     (only used by tests)
+ * @return {?google.maps.drawing.DrawingManager} DrawingManager added to map
+ *     (only used by tests). Null if there was an error
  */
-function initializeAndProcessUserRegions(map, firebasePromise) {
+async function initializeAndProcessUserRegions(map, firebasePromise) {
   addLoadingElement(mapContainerId);
-  return firebasePromise
-      .then((doc) => {
-        // Damage asset may not exist yet, so this is undefined. We tolerate
-        // gracefully.
-        damageAsset = doc.data()['asset_data']['damage_asset_path'];
-        userShapes = getFirestoreRoot().collection(collectionName);
-      })
-      .then(() => userShapes.get())
-      .then(
-          (querySnapshot) => drawRegionsFromFirestoreQuery(querySnapshot, map))
-      .catch(createError('getting user-drawn regions'));
+  try {
+    setUpPopup();
+    let doc;
+    try {
+      doc = await firebasePromise;
+    } catch (err) {
+      // Firebase retrieval error handled elsewhere.
+      throw err;
```

Comment:
Sorry if this is totally obvious, how is catching an error and immediately throwing it again different from just not catching the err?

Reply:
It isn't :) removed in favor of a comment.